scripts/generate_collection_botmeta.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# generate_collection_botmeta.py - a handy script to make botmeta files
+#
+# DESCRIPTION:
+#   Combining existing botmeta via ansibot's code and with the nwo
+#   scenario files, this script will emit appropriate BOTMETA files
+#   for the various repos that nwo specifies
+# 
+# USAGE:
+#   PYTHONPATH=. ./scripts/generate_collection_botmeta.py
+#
+
+import base64
+import copy
+import fnmatch
+import json
+#import logging
+import os
+import re
+import shutil
+import sys
+
+import q
+import yaml  # pyyaml
+
+from collections import OrderedDict
+from collections import defaultdict
+from functools import lru_cache
+
+from github import Github  # pygithub
+from logzero import logger
+import ruamel.yaml
+
+import ansibullbot.constants as C
+from ansibullbot.utils.component_tools import AnsibleComponentMatcher
+from ansibullbot.utils.git_tools import GitRepoWrapper
+from ansibullbot.utils.logs import set_logger
+from ansibullbot.utils.botmeta import botmeta_list
+
+set_logger(debug=True)
+
+
+GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+if GITHUB_TOKEN is None:
+    GITHUB_TOKEN = C.DEFAULT_GITHUB_TOKEN
+assert GITHUB_TOKEN, "GITHUB_TOKEN env var must be set to an oauth token with repo:read from https://github.com/settings/"
```

Comment:
FYI `assert`s are stripped off and aren't evaluated in runtime when `python` is run in optimized mode.